modules/history/src/views/full/index.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react'
+import style from './style.scss'
+
+import { MessagesViewer } from './message'
+import { ConversationPicker } from './conversation'
+
+export default class FullView extends React.Component {
+  constructor(props) {
+    super(props)
+
+    const blob = new Blob([''], { type: 'application/json' })
+    const url = window.URL.createObjectURL(blob)
+
+    const defaultToDate = new Date(Date.now())
+    defaultToDate.setHours(0, 0, 0, 0)
+
+    this.state = {
+      conversationsInfo: [],
+      messageGroups: [],
+      to: defaultToDate,
+      from: this.offsetDateByDays(defaultToDate, -30),
+      currentConvId: null,
+      fileBlob: blob,
+      fileURL: url
+    }
+  }
+
+  threadIdParamName = 'threadId'
+
+  offsetDateByDays(date, offset) {
+    const newDate = new Date(date)
+    newDate.setDate(newDate.getDate() + offset)
+    return newDate
+  }
+
+  componentDidMount() {
+    this.getConversations(this.state.from, this.state.to)
+    const url = new URL(window.location.href)
+    const threadId = url.searchParams.get(this.threadIdParamName)
+    if (threadId) {
+      this.setState({ currentConvId: threadId })
+      this.getMessagesOfConversation(threadId)
+    }
+  }
+
+  getConversations(from, to) {
+    const ceiledToDate = this.offsetDateByDays(to, 1)
+
+    this.props.bp.axios
+      .get(`/mod/history/conversations/${from.getTime()}/${ceiledToDate.getTime()}`)
+      .then(({ data }) => {
+        this.setState({ conversationsInfo: data })
+      })
+  }
+
+  onConversationSelected(convId) {
+    const url = new URL(window.location.href)
+    url.searchParams.set(this.threadIdParamName, convId)
+    window.history.pushState(window.history.state, '', url.toString())
+
+    this.setState({ currentConvId: convId })
+
+    this.getMessagesOfConversation(convId)
```

Comment:
you would usually call `getMessagesOfConversation` on props updated in a sub-component. Doing it this way will help you deal with potential race conditions etc

e.g.

```jsx
<div>
<ConversationsList onChanged={id => this.setState({ convId: id })} />
<ConversationHistory id={this.state.convId} />
</div>
```